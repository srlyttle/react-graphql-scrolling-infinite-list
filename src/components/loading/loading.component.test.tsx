import React from 'react';

import { render } from '@testing-library/react';
import Loading from './loading.component';

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };
test('<Loading is rendered correctly', async () => {
  const loadingMessage = 'I am loading';
  const { findByText, container } = render(
    <Loading large={false} message={loadingMessage} />
  );

  const item = await findByText(loadingMessage);
  expect(item.innerHTML).toBe(loadingMessage);
  expect(container).toMatchSnapshot();
});
