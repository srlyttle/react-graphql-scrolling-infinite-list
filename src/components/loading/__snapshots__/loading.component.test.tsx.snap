// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading is rendered correctly 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-24 ant-col-offset-12"
    >
      <div
        class="ant-spin ant-spin-spinning ant-spin-show-text"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
        <div
          class="ant-spin-text"
        >
          I am loading
        </div>
      </div>
    </div>
  </div>
</div>
`;
